src/testers.jl

```diff
@@ -230,7 +230,16 @@ function test_rrule(
                     "The pullback in the rrule should use NoTangent()" *
                     " rather than ZeroTangent() for non-perturbable arguments.",
                 )
-                @test ad_cotangent isa NoTangent  # we said it wasn't differentiable.
+                if ad_cotangent isa ChainRulesCore.NotImplemented
+                    # this situation can occur if a cotangent is not implemented and
+                    # the default `rand_tangent` is `NoTangent`: e.g. due to having no fields
+                    # the `@test_broken` bellow should tell them that their is an easy
```

Comment:
```suggestion
                    # the `@test_broken` below should tell them that there is an easy
```

Reply:
what do you mean by "there is an easy implementation"?

Reply:
The correct implementation of anything that `test_rrule` determined was a `NoTangent` is almost certainly `NoTangent()`
Which is easy.
It might not generalize to cases that are not tested, but for the types passed into this test it is (almost certainly) correct.

Reply:
I fixed the spelling mistakes in the latest commit (I had already some major changes in my local branch so I just fixed it there).